accepted/00075-live-patching-enhancements.md

```diff
@@ -0,0 +1,189 @@
+
+# Detailed design
+[design]: #detailed-design
+
+This design provides solutions for two separate problems regarding to kernel live patching:
+
+1. [Live patching EOL information](#live-patching-eol-information)
+2. [Live patching CLM workflow](#live-patching-clm-workflow)
+
+## 1. Live patching EOL information
+
+SUSE provides lifecycle information on any product to users via 'zypper lifecycle' plugin. The actual data about a product is provided in a plain CSV file in a supplementary package per product. In case of live patching, this package is called `lifecycle-data-sle-module-live-patching` and is installed together with the product. Zypper lifecycle plugin reads this data to display human-friendly information to users.
+
+Uyuni shall use this information to provide the following:
+ - Display EOL dates for the current live patch in system details
+ - Display an out-of-support systems list in the home page
+ - Use the notifications system and/or send emails to report live patched systems approaching/past EOL
+
+### Data Retrieval
+
+This section evaluates three alternative implementations for this feature.
+
+#### Call 'zypper lifecycle' on client
+
+Uyuni calls `zypper lifecycle` on the client and collects the data via Salt during package profile update. The data is then stored in the database per client along with the kernel live version data.
+
+**Pros:**
+ - `zypper lifecycle` is publicly supported
+ - Can be called directly for a package name (a specific live-patch package)
+ - Can be attached to profile update data with an existing process
+
+**Cons:**
+ - Is a 2-step process that requires `zypper lifecycle` to output to a file
+ - `zypper-lifecycle-plugin` needs to be installed on the client
+ - Requires a round-trip to the client, per client
+
+#### Read the lifecycle data file on client
+
+Uyuni runs a Salt state to read the lifecycle data file provided with the package `lifecycle-data-sle-module-live-patching` on the client and collects the data during package profile update. The data is then stored in the database per client along with the kernel live version data.
+
+**Pros:**
+ - Easy to implement
+ - Can be attached to profile update data with an existing process
+
+**Cons:**
+ - File name or format can change in the future
+ - Requires a round-trip to the client, per client
+
+#### Unpack and read the lifecycle data RPM on Uyuni server
+
+The lifecycle information is provided in the `lifecycle-data-sle-module-live-patching` package in the Live Patching product. Uyuni unpacks this RPM and reads the data from the extracted CSV file during reposync of this product. The data is stored in the database per live patch package.
+
+A high-level algorithm of the process is as follows:
+```
+# At reposync or as a separate task after sync
+
+for all RPMs that start with `lifecycle-data-*`:
```

Comment:
This is bad, what happens when they decide to change the file name or for a specific product put it into a different package. We should not rely on the package name.

When we parse a package, we extract the filelist from the metadata and put every file in the database.
So we could simply query every file which is in a directory and find the corresponding package.

Reply:
We could parse every package and look for `*.lifecycle` files everywhere, but I'm not sure if it will have some significant performance burden

Reply:
@cbbayburt we already extract the whole filelist and save it into the DB, so just querying it after the fact will be fine.